lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -433,6 +423,19 @@ def validate_selector(selector)
       end
     end
 
+    def validate_spec_with_kubectl(kubectl)
+      command = ["create", "-f", file_path, "--dry-run", "--output=name"]
+      _, err, st = kubectl.run(*command, log_failure: false, output_is_sensitive: sensitive_template_content?,
+        retry_whitelist: [Kubectl::ERROR_MATCHERS[:client_timeout]], attempts: 3)
```

Comment:
It feels strange to have to pass a value of a hash. I think `retry_whitelist: [:client_timeout]` or `[Kubectl::ERROR_MATCHERS::CLIENT_TIMEOUT]` seem more natural. But its a minor thing.

Reply:
Good idea, I'll make the caller pass a symbol and the class will internally look it up in `ERROR_MATCHERS`